tests/CMakeLists.txt

```diff
@@ -4,3 +4,18 @@ TARGET_INCLUDE_DIRECTORIES(test_edge_sensor PRIVATE ${EDGE_REQUIRE_PKGS_INCLUDE_
 TARGET_LINK_LIBRARIES(test_edge_sensor -L${PROJECT_BINARY_DIR}/src edge-sensor)
 
 INSTALL (TARGETS test_edge_sensor DESTINATION ${BIN_INSTALL_DIR})
+
+# GoogleTest requires at least C++11
+SET(CMAKE_CXX_STANDARD 11)
+
+INCLUDE(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
+)
+
+FetchContent_MakeAvailable(googletest)
+
+ADD_EXECUTABLE(unittest_edge_sensor unittest_edge_sensor.cc)
+TARGET_INCLUDE_DIRECTORIES(unittest_edge_sensor PRIVATE ${INCLUDE_DIR})
+TARGET_LINK_LIBRARIES(unittest_edge_sensor gtest pthread edge-sensor)
```

Comment:
Instead of using the linked library names directly, what about this?
```cmake
TARGET_LINK_LIBRARIES(unittest_edge_sensor ${TEST_REQUIRE_PKGS_LDFLAGS} edge-sensor)
```
To use this way, you need to update the `CMakefiles.txt` as below.
```cmake
  SET(TEST_REQUIRES_LIST
      gtest
  )
  INCLUDE(FindPkgConfig)
  PKG_CHECK_MODULES(TEST_REQUIRE_PKGS REQUIRED ${TEST_REQUIRES_LIST})
```

In case of Tizen, `BuildRequires:  gtest-devel` in spec is required.

Reply:
Thanks! Updated.